src/popup/router/components/Modals/ConfirmTransactionSign.vue

```diff
@@ -130,45 +162,146 @@ export default {
+    ...mapState('fungibleTokens', ['availableTokens']),
+    ...mapGetters(['formatCurrency', 'account', 'activeNetwork']),
+    ...mapGetters([
+      'getTxSymbol',
+      'getTxAmountTotal',
+      'getTxFee',
+      'getNameFee',
+      'isTxAex9',
+      'getDexContracts',
+      'getTxDirection',
+    ]),
+    isAllowance() {
+      return FUNCTION_TYPE_DEX.allowance.includes(this.txFunction);
+    },
+    isSwap() {
+      return FUNCTION_TYPE_DEX.swap.includes(this.txFunction);
+    },
+    isPool() {
+      return FUNCTION_TYPE_DEX.pool.includes(this.txFunction);
+    },
+    isDex() {
+      return this.getDexContracts.router.includes(this.transaction?.contractId)
+        || this.getDexContracts.wae.includes(this.transaction?.contractId)
+        || this.isAllowance;
+    },
+    singleToken() {
+      return {
+        isReceived: this.getTxDirection(this.transaction) === 'received',
+        amount: this.getTxAmountTotal(this.transaction),
+        symbol: this.getTxSymbol(this.transaction),
+        isAex9: this.isTxAex9(this.transaction),
+        preventAmountConversion: true,
+        showFiat: true,
+      };
+    },
+    filteredTxFields() {
+      return this.TX_FIELDS.filter((field) => !!this.transaction[field]);
+    },
+    swapDirection() {
+      if (['swap_tokens_for_exact_tokens', 'swap_tokens_for_exact_ae', 'swap_ae_for_exact_tokens'].includes(this.txFunction)) return 'minReceived';
+      if (['swap_exact_tokens_for_tokens', 'swap_exact_ae_for_tokens', 'swap_exact_tokens_for_ae'].includes(this.txFunction)) return 'maxSpent';
+      return '';
+    },
+    completeTransaction() {
+      return { ...this.transaction, function: this.txFunction };
+    },
+  },
+  async mounted() {
+    if (this.transaction.contractId) {
+      try {
+        this.loading = true;
+        setTimeout(() => { this.loading = false; }, 20000);
+        await this.$watchUntilTruly(() => this.sdk);
+        const { bytecode } = await fetchJson(`${this.activeNetwork.url}/v3/contracts/${this.transaction.contractId}/code`);
+        const txParams = await this.sdk.compilerApi.decodeCalldataBytecode({
+          bytecode,
+          calldata: this.transaction.callData,
+        });
+        this.txFunction = txParams.function;
+        const allTokens = this.getTokens(txParams);
+
+        this.tokenList = allTokens.map((token) => ({
+          ...token,
+          tokens: token.isPool ? allTokens.filter((t) => !t.isPool).reverse() : [token],
+        }));
+      } catch (e) {
+        this.tokenList = null;
+        this.txFunction = null;
+      } finally {
+        this.loading = false;
+      }
+    }
+  },
+  methods: {
+    getTokens(txParams) {
+      if (!this.isDex) return [this.singleToken];
+      const resolver = transactionTokenInfoResolvers[camelCase(txParams.function)];
+      if (!resolver) return [];
+      const tokens = resolver(
+        { tx: { ...txParams, ...this.transaction } }, this.availableTokens,
```

Comment:
**fix**: there was not enough info for transactions: Allowance and AE related
![image](https://user-images.githubusercontent.com/9008074/181733917-b4b8dbd8-82f1-4fdd-9e53-c40985835bec.png)
![image](https://user-images.githubusercontent.com/9008074/181733936-c3882915-6478-48ca-a7df-e007f9d04cda.png)